raster/r.buffer/testsuite/test_buffer.py

```diff
@@ -0,0 +1,126 @@
+from grass.gunittest.case import TestCase
+from grass.gunittest.main import test
+from grass.gunittest.gmodules import SimpleModule
+import grass.script as gs
+
+
+class TestRBuffer(TestCase):
+
+    @classmethod
+    def setUpClass(cls):
+        cls.use_temp_region()
+        cls.runModule("g.region", raster="roadsmajor")
+
+    @classmethod
+    def tearDownClass(cls):
+        cls.del_temp_region()
+
+    def tearDown(self):
+        # Remove temporary maps created during tests
+        gs.run_command(
+            "g.remove",
+            type="raster",
+            name="null_map,zero_map,buf_test,buf_no_non_null,buf_ignore_zero",
+            flags="f",
+        )
+
+    def test_buffer_creation(self):
+        output = "buf_test"
+        distances = [100, 200, 300, 400, 500]
+
+        module = SimpleModule(
+            "r.buffer",
+            input="roadsmajor",
+            output=output,
+            distances=distances,
+            overwrite=True,
+        )
+        self.assertModule(module)
+
+        self.assertRasterExists(output)
+
+        expected_categories = [1] + [i + 1 for i in range(len(distances) + 1)]
+
+        self.assertRasterMinMax(
+            map=output,
+            refmin=min(expected_categories),
+            refmax=max(expected_categories),
+            msg=(
+                "Buffer zones should have category values from 1 to "  # Checking if there are no abnormal values in the output raster map
+                + str(max(expected_categories))
+            ),
+        )
+
+        category_values = gs.read_command(
+            "r.stats", flags="n", input=output
+        ).splitlines()
+        category_values = [int(line.split()[0]) for line in category_values]
+
+        print("Category values:", category_values)
+
+        unique_actual_categories = set(category_values)
+        self.assertTrue(
+            unique_actual_categories.issubset(set(expected_categories)),
+            msg=(
+                "Output categories should be a subset of expected categories: "
+                + str(expected_categories)
+            ),
+        )
+
+    def test_no_non_null_values(self):
+        null_map = "null_map"
+        self.runModule("r.mapcalc", expression=f"{null_map} = null()")
+
+        output = "buf_no_non_null"
+        distances = [100, 200, 300]
+
+        module = SimpleModule(
+            "r.buffer",
+            input=null_map,
+            output=output,
+            distances=distances,
+            overwrite=True,
+        )
+        self.assertModule(module)
+
+        self.assertRasterExists(output)
+        stats = gs.read_command("r.univar", map=output, flags="g")
+        self.assertIn(
```

Comment:
https://grass.osgeo.org/grass83/manuals/libpython/gunittest.html#gunittest.case.TestCase.assertRasterFitsUnivar

Reply:
In this test I use r.univar to ensure that the output map has no non-null cells when the input is entirely null. According to my understanding r.univar helps calculate univar stats and if it is a null output it gives out n=0 (number of non null cells). This verifies that the buffer operation correctly handles null input data without introducing any non-null values.

Reply:
Used assertRasterFitsUnivar instead of manually doing it